extension/src/content/index.js

```diff
@@ -46,6 +45,42 @@ function init() {
     yaml: "yaml"
   };
 
+  // Pasting events are used to force stackoverflows WMD to save state.
+  // TODO: This solution is buggy and does not always work - more discussion needed on solution
+  function stackOverflowFormat(
+    event,
+    textAreaElem,
+    text = null,
+    parser = "markdown"
+  ) {
+    event.preventDefault();
+    textAreaElem.dispatchEvent(new Event("paste"));
+    textAreaElem.input = formatText(event, textAreaElem, text, parser);
+    textAreaElem.dispatchEvent(new Event("paste"));
+  }
+
+  function formatText(event, textAreaElem, text = null, parser = "markdown") {
```

Comment:
This function has a really complex/confusing signature for what it's doing, which I think is a code smell indicating it might be doing too much. I actually think it's simpler to have the GitHub and StackOverflow formatting functions be separated out for now (my plan is to move them into separate modules once we land this) rather than coupled like this. We can always refactor and create abstractions later as we find similar code. Thoughts?

Reply:
I do think it makes sense to potentially share code for the actual DOM mutation though, since there is some shared behavior around focusing the text area, etc.